.github/workflows/test.yml

```diff
@@ -63,3 +63,55 @@ jobs:
 
       - name: Run unit tests
         run: composer phpunit
+
+  coverage:
+    needs: test
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        php: [ '7.4' ]
+        experimental: [ false ]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          ini-values: zend.assertions=1, error_reporting=-1, display_errors=On
+          coverage: xdebug
+
+      - name: Install Composer dependencies - normal
+        uses: "ramsey/composer-install@v2"
+
+      - name: Run the unit tests with code coverage
+        run: vendor/bin/phpunit
+
+      - name: Install Coveralls
+        if: ${{ success() }}
+        run: composer require php-coveralls/php-coveralls:"^2.5.2" --no-interaction
+
+      - name: Upload coverage results to Coveralls
+        if: ${{ success() }}
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_PARALLEL: true
+          COVERALLS_FLAG_NAME: php-${{ matrix.php }}-phpcs-${{ matrix.phpcs_version }}
```

Comment:
As you are only running one build (PHP 7.4) for code coverage, no need to set the "parallel" flag or to name the builds as for each push/PR, there will only ever be one.

Also see: https://github.com/php-coveralls/php-coveralls#parallel-builds

This also means that the complete `coveralls-finish` job shouldn't be needed.